### PR TITLE
Use catch2 version 3.7.1

### PR DESCRIPTION
**Pull request by arcondello**
The latest version supported by meson wrap as of Sept 2024.

Now that meson is handling install/build of catch2 for us. It makes sense to use the latest. We were on quite an old branch.

Also fix some compiler warnings generated by `IntegerNode`.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -97,7 +97,7 @@ jobs:
             . env/bin/activate
             python -m build \
               -Ccompile-args="-v" \
-              -Csetup-args="-Dbuildtype=debug" \
+              -Csetup-args="-Db_ndebug=true" \
```
Comment (arcondello) [inline .circleci/config.yml:100]: the `bdebug` build type builds the C++ test, which doesn't work with creating a wheel. So we instead turn on debug symbols directly.

File: tests/cpp/tests/test_array.cpp [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 #include <sstream>
 
-#include "catch2/catch.hpp"
+#include "catch2/catch_all.hpp"
```
Comment (wbernoudy) [inline tests/cpp/tests/test_array.cpp:17]: Isn't the `catch_all.hpp` header significantly bigger than the original `catch2/catch.hpp` header? Is it slower to compile now?

Comment (wbernoudy) [reply tests/cpp/tests/test_array.cpp:17]: I see they recommend doing this replacement in the upgrade docs so I'm sure it's fine. Just curious.

Comment (arcondello) [reply tests/cpp/tests/test_array.cpp]: Ok. I changed to use more specific headers. Though the compilation time changed not at all so :shrug: .

Comment (wbernoudy) [reply tests/cpp/tests/test_array.cpp]: Disappointing!

**Review by wbernoudy — APPROVED** [review]

**Pull request merged**